### PR TITLE
[libcxxabi][libunwind] Support for using LLVM libc

### PR DESCRIPTION
**Pull request by petrhosek**
This is analogous to #99287 and provides an option to build libc++abi and libunwind against LLVM libc when selected.

### CODE REVIEW

**Comment by petrhosek** [comment]
@ldionne I'm starting to think that we should have a generic module in https://github.com/llvm/llvm-project/tree/main/cmake/Modules that would be controlled by `LLVM_USE_LIBC` and define `libc-headers`, `libc-static` and `libc-shared`. That way we could avoid duplicating this logic in every subproject. What do you think?

File: libunwind/cmake/Modules/HandleLibC.cmake [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    target_link_libraries(libunwind-libc-static INTERFACE libc)
+  endif()
+  if (TARGET libm)
+    target_link_libraries(libunwind-libc-static INTERFACE libm)
+  endif()
+  if (CXX_SUPPORTS_NOLIBC_FLAG)
+    target_link_options(libunwind-libc-static INTERFACE "-nolibc")
+  endif()
+
+  # TODO: There's no support for building LLVM libc as a shared library yet.
```
Comment (SchrodingerZhu) [inline libunwind/cmake/Modules/HandleLibC.cmake:37]: I was about to mention that but then I saw this comment.

**Comment by ldionne** [comment]
> @ldionne I'm starting to think that we should have a generic module in https://github.com/llvm/llvm-project/tree/main/cmake/Modules that would be controlled by `LLVM_USE_LIBC` and define `libc-headers`, `libc-static` and `libc-shared`. That way we could avoid duplicating this logic in every subproject. What do you think?

I agree, but I might put it in `runtimes/cmake/Modules` where we already have some stuff. I also wonder if we want to use  `LLVM_USE_LIBC` or `RUNTIMES_USE_LIBC` or something else. Historically, `LLVM_FOO` settings are mostly things that we've inherited from `<monorepo>/llvm/CMakeLists.txt` and tried to move away from.

File: libcxxabi/cmake/Modules/HandleLibC.cmake [hunk 2]
```diff
@@ -0,0 +1,39 @@
+#===============================================================================
```
Comment (ldionne) [inline libcxxabi/cmake/Modules/HandleLibC.cmake:1]: For now, can't you include `libcxx/cmake/Modules/HandleLibC.cmake` instead?

**Comment by ldionne** [comment]
Gentle ping

**Comment by ldionne** [comment]
Closing as stale, please reopen if you still want to pursue.

**Pull request closed without merging**